packages/browser/src/index.ts

```diff
@@ -17,39 +34,235 @@ export type { SignerError, SignerErrorCode } from './-private/errors/signer-erro
+   */
+  public getConnectButton(
+    options?: ConnectButtonOptions,
+    callback?: (error?: Error, user?: any) => void,
+  ): ConnectButton {
+    return new ConnectButton(this.sdk, options, callback);
+  }
+
+  /**
+   * Signs in or connects to bitski depending on the user's auth state.
+   * Since it may open a popup, this method must be called from user interaction handler,
+   * such as a click or tap handler.
+   * @param options Provide SignInOptions for the sign in request. See signIn() for more info.
+   */
+  public async start(options?: SignInOptions): Promise<User> {
+    return this.getSDK().then((sdk) => sdk.signInOrConnect(undefined, options));
+  }
 
-/**
- * Alternative to using our static callback.html file. Call this from your own redirect page.
- */
-export const popupCallback = processCallback;
+  /**
+   * Check the logged in state of the user
+   */
+  public async getAuthStatus(): Promise<AuthenticationStatus> {
+    return this.getSDK().then((sdk) => sdk.getAuthStatus());
+  }
+
+  /**
+   * Starts the sign in flow. Will trigger a popup window over your app, so it must be called within a user interaction handler such as a click.
+   * @param options Optionally provide additional options for the sign in request.
+   *
+   * You can use the options parameter to request that we show the sign up form instead of the sign in form:
+   * ```javascript
+   * import { LOGIN_HINT_SIGNUP } from 'bitski';
+   *
+   * await bitski.signIn({ login_hint: LOGIN_HINT_SIGNUP });
+   * ```
+   */
+  public async signIn(options?: SignInOptions): Promise<User> {
+    return this.getSDK().then((sdk) => sdk.signIn(options));
+  }
+
+  /**
+   * Gets the current signed in user. Will reject if we are not signed in.
+   */
+  public async getUser(): Promise<User> {
+    return this.getSDK().then((sdk) => sdk.getUser());
+  }
+
+  /**
+   * Connects to bitski to get a valid access token if possible.
+   */
+  public async connect(): Promise<User> {
+    return this.getSDK().then((sdk) => sdk.connect());
+  }
+
+  /**
+   * Starts redirect sign in flow. This is an alternative flow to the popup that all takes place in the same browser window.
+   * @param options Optionally provide additional options for the sign in request. See signIn() for more info.
+   */
+  public signInRedirect(options?: SignInOptions): void {
+    this.getSDK().then((sdk) => sdk.signInRedirect(options));
+  }
+
+  /**
+   * Call from your oauth redirect page.
+   */
+  public async redirectCallback(): Promise<User> {
+    return this.getSDK().then((sdk) => sdk.redirectCallback());
+  }
+
+  /**
+   * Retrieves the current access token for the user, if logged in.
+   */
+  public async getCurrentAccessToken(): Promise<string> {
+    return this.getSDK().then((sdk) => sdk.getCurrentAccessToken());
+  }
+
+  /**
+   * Retrieves the current ID token for the user, if logged in.
+   */
+  public getCurrentIdToken(): Promise<string | undefined> {
```

Comment:
What's the id token?

Reply:
it's the oauth id token, I'm just re-exposing the existing implementation here